Sprint 5

UlyanovVS opened the pull request:
1. Classes:
 - Update: Cat, Feline, Lion, Predator
 - Add: LionAlex
2. Tests:
 - Add: CatParameterizedTest, CatTest, FelineParameterizedTest, FelineTest, LionAlexParameterizedTest, LionAlexTest, LionParameterizedTest, LionTest
3. Update: pom.xml
4. Add: jacoco report

Commit:
1. Classes:
 - Update: Cat, Feline, Lion, Predator
 - Add: LionAlex
2. Tests:
 - Add: CatParameterizedTest, CatTest, FelineParameterizedTest, FelineTest, LionAlexParameterizedTest, LionAlexTest, LionParameterizedTest, LionTest
3. Update: pom.xml

Commit:
add jacoco report

mkouznetsov-git commented on src/main/java/com/example/Lion.java:
```diff
@@ -3,30 +3,29 @@
-    boolean hasMane;
-
-    public Lion(String sex) throws Exception {
+    public Lion(String sex, Predator predator) throws Exception {
```
Вместо интерфейса Predator следует использовать Feline, т.к. это более точно описывает поведение льва. Класс кошачьих как раз подходит, интерфейс Хищник слишком общее, там может быть что угодно (акула, ястреб, змея и т.п.) -- зачем это в классе льва? )

mkouznetsov-git commented on src/test/java/CatParameterizedTest.java:
```diff
@@ -0,0 +1,15 @@
+package com.example;
```
Некорректно указан пакет

mkouznetsov-git commented on src/test/java/CatTest.java:
```diff
@@ -0,0 +1,23 @@
+package com.example;
```
Некорректно указан пакет

mkouznetsov-git commented on src/test/java/CatParameterizedTest.java:
```diff
@@ -0,0 +1,15 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class CatParameterizedTest {
```
Чем принципиально этот класс отличается от CatTest? Проверка одна и та же. Это лишне дублирование кода, оставь что-то одно.

mkouznetsov-git commented on src/test/java/LionAlexParameterizedTest.java:
```diff
@@ -0,0 +1,27 @@
+class LionAlexParameterizedTest {
+
+    @ParameterizedTest
+    @CsvSource({
```
Проверка на гриву есть в тестах на класс льва. Здесь это лишнее

mkouznetsov-git commented on src/test/java/LionTest.java:
```diff
@@ -0,0 +1,52 @@
+    Predator predator;
+
+    @Test
+    void shouldHaveManeIfMale() throws Exception {
```
Этот тест есть в параметризованном классе. Здесь он не нужен.

mkouznetsov-git commented on src/test/java/LionTest.java:
```diff
@@ -0,0 +1,52 @@
+    }
+
+    @Test
+    void shouldNotHaveManeIfFemale() throws Exception {
```
Этот тест есть в параметризованном классе. Здесь он не нужен.

Commit:
1. Update:
- classes: Feline, Lion, LionAlex
- interface: Predator
- tests: CatTest, FelineParameterizedTest, FelineTest, LionAlexTest, LionParameterizedTest, LionTest
2. Delete: CatParameterizedTest, LionAlexParameterizedTest

Commit:
add jacoco report 2
